test/GoodGhostingGasEstimate.test.js

```diff
@@ -34,11 +43,24 @@ contract("GoodGhosting", (accounts) => {
     describe("simulates a full game with 5 players and 4 of them winning the game and with admin fee % as 0", async () => {
         it("initializes contract instances and transfers DAI to players", async () => {
             token = new web3.eth.Contract(daiABI, providersConfigs.dai.address);
-            const poolConfigs = configs.providers[configs.deployConfigs.selectedProvider.toLowerCase()]['mainnet'];
-            const lendingPoolAddressProvider = poolConfigs.lendingPoolAddressProvider;
-            const dataProviderAddress = poolConfigs.dataProvider;
-            const inboundCurrencyAddress = poolConfigs[configs.deployConfigs.inboundCurrencySymbol.toLowerCase()].address;
-            goodGhosting = await GoodGhosting.new(inboundCurrencyAddress, lendingPoolAddressProvider, 4, 180, segmentPayment, 10, 0, dataProviderAddress, "0xd566243e283f1357e5e97dd0c9ab0d78177583074b440cb07815e05f615178bf");
+            // overriding deployment to test with 0% custom fees
+            if (process.env.NETWORK === "local-mainnet-fork") {
+                const poolConfigs = configs.providers[configs.deployConfigs.selectedProvider.toLowerCase()]['mainnet'];
+                const lendingPoolAddressProvider = poolConfigs.lendingPoolAddressProvider;
+                const dataProviderAddress = poolConfigs.dataProvider;
+                const inboundCurrencyAddress = poolConfigs[configs.deployConfigs.inboundCurrencySymbol.toLowerCase()].address;
+                goodGhosting = await GoodGhostingArtifact.new(inboundCurrencyAddress, lendingPoolAddressProvider, segmentCount, segmentLength, segmentPayment, earlyWithdrawFee, 0, dataProviderAddress, configs.deployConfigs.merkelroot);
+            } else {
+                const poolConfigs = configs.providers[configs.deployConfigs.selectedProvider.toLowerCase()]['polygon'];
+                const lendingPoolAddressProvider = poolConfigs.lendingPoolAddressProvider;
+                const incentiveController = poolConfigs.incentiveController;
+                const router = poolConfigs.router;
+                const usdc = poolConfigs.usdc;
+                const wmatic = poolConfigs.wmatic;
+                const dataProviderAddress = poolConfigs.dataProvider;
+                const inboundCurrencyAddress = poolConfigs[configs.deployConfigs.inboundCurrencySymbol.toLowerCase()].address;
+                goodGhosting = await GoodGhostingArtifact.new(inboundCurrencyAddress, lendingPoolAddressProvider, segmentCount, segmentLength, segmentPayment, earlyWithdrawFee, 0, dataProviderAddress, configs.deployConfigs.merkelroot, incentiveController, router, wmatic, usdc);
+            }
```

Comment:
not sure why we need to revert this back instead of using the previous suggested approach (`GoodGhostingArtifact.deployed()`).
By using the previous option, we just get the deployed contract (based on the deploy-configs.js file) and run the tests on those parameters. This allows us to test both games with and without admin fees, instead of hardcoding 0 admin fees. Considering this contract is used for gas estimation, it would be good to estimate how much gas is used for withdrawing the admin fees (when fee is > 0), rather than not having that info available.

Reply:
yeah will revert this